GeneralsMD/Code/GameEngine/Source/GameLogic/Object/Weapon.cpp

```diff
@@ -1190,6 +1191,21 @@ void WeaponTemplate::trimOldHistoricDamage() const
 		}
 	}
 }
+#else
+void WeaponTemplate::trimOldHistoricDamage() const
+{
+	HistoricWeaponDamageList::iterator it = m_historicDamage.begin();
+
+	while (it != m_historicDamage.end())
+	{
+		UnsignedInt expirationDate = it->frame + m_historicBonusTime;
+		if (TheGameLogic->getFrame() > expirationDate || it->triggered)
+			it = m_historicDamage.erase(it);
+		else
+			++it;
```

Comment:
Is the chronological order no longer present? If it still is chronological , then aborting on first fail is more efficient.